cfme/tests/containers/test_smart_management.py

```diff
@@ -0,0 +1,106 @@
+from cfme.containers.provider import ContainersProvider, ContainersTestItem
+from cfme.containers.image import Image
+from cfme.containers.project import Project
+from cfme.containers.node import Node
+from cfme.containers.image_registry import ImageRegistry
+from cfme.containers.pod import Pod
+from cfme.containers.template import Template
+from cfme.containers.provider import navigate_and_get_rows
+
+pytestmark = [
+    pytest.mark.uncollectif(lambda: current_version() < "5.6"),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(1)]
+pytest_generate_tests = testgen.generate([ContainersProvider], scope='function')
+
+TEST_ITEMS = [
+    pytest.mark.polarion('CMP-10320')(ContainersTestItem(Template, 'CMP-10320')),
+    pytest.mark.polarion('CMP-9992')(ContainersTestItem(ImageRegistry, 'CMP-9992')),
+    pytest.mark.polarion('CMP-9981')(ContainersTestItem(Image, 'CMP-9981')),
+    pytest.mark.polarion('CMP-9964')(ContainersTestItem(Node, 'CMP-9964')),
+    pytest.mark.polarion('CMP-9932')(ContainersTestItem(Pod, 'CMP-9932')),
+    pytest.mark.polarion('CMP-9870')(ContainersTestItem(Project, 'CMP-9870')),
+    pytest.mark.polarion('CMP-9854')(ContainersTestItem(ContainersProvider, 'CMP-9854'))
+]
+
+
+def set_random_tag(instance):
+    navigate_to(instance, 'Details')
+    toolbar.select('Policy', 'Edit Tags')
+
+    # select random tag category
+    cat_selector = AngularSelect("tag_cat")
+    random_cat = random.choice(cat_selector.all_options)
+    cat_selector.select_by_value(random_cat.value)
+
+    # select random tag tag
+    tag_selector = AngularSelect("tag_add")
+    random_tag = random.choice([op for op in tag_selector.all_options if op.value != "select"])
+    tag_selector.select_by_value(random_tag.value)
+
+    # Save tag conig
+    form_buttons.save()
+
+    return Tag(display_name=random_tag.text, category=random_cat.text)
+
+
+def obj_factory(obj_creator, row, provider):
+
+    factory = {"Provider": lambda row: {"name": row.name.text},
+               "Project": lambda row: {"name": row.name.text, "provider": provider},
+               "Pod": lambda row: {"name": row.name.text, "provider": provider},
+               "Node": lambda row: {"name": row.name.text, "provider": provider},
+               "Template": lambda row: {"name": row.name.text, "provider": provider},
+               "Image": lambda row: {"name": row.name.text,
+                                     "tag": row.tag.text, "provider": provider},
+               "Image_Registry": lambda row: {"host": row.host.text, "provider": provider}}
+
+    return obj_creator(**factory[obj_creator.__module__.title().split(".")[-1]](row))
+
+
+@pytest.mark.parametrize('test_item',
+                         TEST_ITEMS, ids=[testItem.args[1].pretty_id() for testItem in TEST_ITEMS])
+def test_smart_management_add_tag(provider, test_item):
+
+    # Select random instanse from instanse table
+    chosen_row = navigate_and_get_rows(provider, test_item.obj, 1).pop()
```

Comment:
navigate_and_get_rows is a function that navigateing to some object page and get few random rows as requested.

In this case i needed only one row, so the function returns list with only one value.
to reduce the amount of code and un-needed variables I decided to pop out the one value on the same line.